docs/guides/upgrading-v3.md

```diff
@@ -0,0 +1,265 @@
+
+The default Lambda runtime has changed from NodeJS 12 to NodeJS 14, given this is now the default runtime recommended by AWS.
+
+In `serverless.yml`, the `service` key no longer accepts a YAML object ([learn more](../deprecations.md#service-property-object-notation)).
+
+```yaml
+# Deprecated in v2, removed in v3:
+service:
+    name: my-service
+
+# Correct syntax:
+service: my-service
+```
+
+All options that used to be defined inside the `service` key have been moved to other sections (mentioned below in that document). This change clears up confusion that existed between the `service` and `provider` sections.
+
+CloudFormation extensions defined in `resources.extensions` are now validated: extensions that point to unknown resources will throw an error, which should help catch broken configuration before deploying ([learn more](../deprecations.md#defining-extensions-to-nonexistent-resources-in-resourcesextensions)).
+
+### API Gateway
+
+When configuring API Gateway, some configuration options have moved to a dedicated sub-section of `provider`. That will help clear up confusion with similar `httpApi` settings.
+
+```yaml
+provider:
+  # Deprecated in v2, removed in v3:
+  apiKeys: ...
+  resourcePolicy: ...
+  usagePlan: ...
+
+  # Correct syntax:
+  apiGateway:
+    apiKeys: ...
+    resourcePolicy: ...
+    usagePlan: ...
+```
+
+The `schema` option on HTTP events has also been renamed to `schemas`. That change allowed us to support much more schema validation features:
+
+```yaml
+functions:
+  hello:
+    handler: hello.handler
+    events:
+      - http:
+        ...
+        request:
+          # Deprecated in v2, removed in v3:
+          schema: ...
+          # Correct syntax:
+          schemas: ...
+```
+
+Learn more about [schema changes](../deprecations.md#aws-api-gateway-schemas).
+
+When an external API Gateway is used and imported via `provider.apiGateway.restApiId`, both `provider.logs.restApi` and `provider.tracing.apiGateway` options will be ignored. These settings are applicable only if the API Gateway is provisioned by Serverless Framework.
+
+The CloudFormation tags defined in `provider.tags` will now be correctly applied to HTTP APIs ([learn more](../deprecations.md#http-api-provider-tags)).
+
+Starting with v3, a bug fix for API Gateway authorizers will become the default behavior (to match AWS’ recommended behavior): `functions[].events[].http.authorizer.identitySource` will no longer be set to `method.request.header.Authorization` by default when caching is disabled (i.e. for authorizers of type "request" with `resultTtlInSeconds` set to "0") ([learn more](../deprecations.md#default-identitysource-for-httpauthorizer)).
+
+### CloudFront
+
+CloudFront’s `behavior` option (ForwardedValues, MinTTL, MaxTTL and DefaultTTL) has been removed, as this feature was deprecated by AWS. Use the new "cache policy" feature instead ([learn more](../deprecations.md#cloudfront-event-behaviorforwardedvalues-property)).
```

Comment:
Wouldn't it be nicer to link `new "cache policy feature instead` to AWS documentation resource,  instead of adding extra "learn more" link?

Reply:
I think the current wording might be a bit confusing, as we still support `behavior` setting, just a few of it's options has been removed (the ones mentioned in parentheses).

Reply:
Good point, I added the link and rephrased it. Let me know if that's good.